Gemfile

```diff
@@ -3,6 +3,10 @@ gemspec
 
 gem 'rake'
 
+# for Ruby 2.4
+gem 'mongo', '< 2.17'
+gem 'bson', '< 4.13'
```

Comment:
Ruby 2.4 requires `bson < 4.13`.
https://github.com/mongodb/bson-ruby/releases/tag/v4.13.0

However, JRuby 9.3 requires 'bson >= 4.14'.
https://github.com/mongodb/bson-ruby/compare/v4.13.0...v4.14.0

Reply:
If so, why mongomapper's CI for both Ruby 2.4 and JRuby 9.3 pass now?

Reply:
Ruby 2.4 installs bson 4.12.
https://github.com/mongomapper/mongomapper/runs/6596360345?check_suite_focus=true#step:5:42

JRuby 9.3 installs bson 4.15.
https://github.com/mongomapper/mongomapper/runs/6596360078?check_suite_focus=true#step:5:40

We don't specify bson's version in the Gemfiles for mongomapper's CI.
Then, bundler picks up for their runtime.